### ISSUE TITLE
Generated dictionary sometimes includes word twice, once followed by a period

### ISSUE THREAD

**Issue by tbroadley** [tbroadley/spellchecker-cli#26]
For example, [here](https://github.com/eslint/eslint/blob/14f1713519f106376c517aee90ef26d8786bfd87/dictionary.txt#L21).

**Comment by tbroadley** [comment]
Minimal example that causes this bug:

```md
misspellig.

* lowercase
```

The following does not trigger the bug:

```md
misspellig.

* Uppercase
```

**Comment by a2937** [comment]
Hmm on the personal tiny spellchecker I made; I just flat out removed most punctuation from words that needed spellchecked. Could we do something like this here?

**Comment by a2937** [comment]
While removing periods, exclamation marks, and question mark nodes from words that need spellchecked ; it could probably be helpful here?

**Comment by tbroadley** [comment]
True, that would solve this problem. I'm hesitant to have spellchecker-cli preprocess text too heavily before passing it into Retext. For example, if we removed punctuation, that might conflict with one or more Retext [plugins](https://github.com/retextjs/retext/blob/main/doc/plugins.md). I have an open issue for adding support for more Retext plugins to spellchecker-cli: #18 

I'd rather resolve this issue by fixing the underlying issue in Retext or `remark-retext`.
